src/editors/playgroundController.ts

```diff
@@ -666,19 +686,20 @@ export default class PlaygroundController {
 
     const selections = this._activeTextEditor.selections;
 
+    let codeToEvaluate;
     if (
       !selections ||
       !Array.isArray(selections) ||
       (selections.length === 1 && this._getSelectedText(selections[0]) === '')
     ) {
       this._isPartialRun = false;
-      this._codeToEvaluate = this._getAllText();
+      codeToEvaluate = this._getAllText();
     } else if (this._selectedText) {
       this._isPartialRun = true;
-      this._codeToEvaluate = this._selectedText;
+      codeToEvaluate = this._selectedText;
     }
 
-    return this._evaluatePlayground();
+    return this._evaluatePlayground(codeToEvaluate);
```

Comment:
[nit] With the current code structure, we may be passing `undefined` here, while `_evaluatePlayground` expects string. My suggestion would be to declare the type of `codeToEvaluate` and either return early in an `else` clause for the `if`-statement on line 690, or fallback to an empty string.